runtime/rasdump/javadump.cpp

```diff
@@ -5149,6 +5164,59 @@ JavaCoreDumpWriter::writeCPUinfo(void)
 	return;
 }
 
+void
+JavaCoreDumpWriter::writeCgroupMetrics(void)
+{
+	OMRPORT_ACCESS_FROM_J9PORT(_PortLibrary);
+	BOOLEAN isCgroupSystemAvailable = omrsysinfo_cgroup_is_system_available();
+	if (isCgroupSystemAvailable) {
+		OMRCgroupEntry *entryhead = omrsysinfo_get_cgroup_subsystem_list();
+		OMRCgroupEntry *cgEntry = entryhead;
+		OMRCgroupEntry *nextEntry = NULL;
+		int32_t rc = 0;
+		char *tag = (char *)"2CICGRPINFO";
+		if (NULL != cgEntry) {
+			_OutputStream.writeCharacters("NULL \n");
+			_OutputStream.writeCharacters("1CICGRPINFO    Cgroup Information \n");
+			_OutputStream.writeCharacters("NULL           ------------------------------------------------------------------------\n");
+			do {
+				nextEntry = cgEntry->next;
+				_OutputStream.writeCharacters("2CICGRPINFO    subsystem : ");
+				_OutputStream.writeCharacters(cgEntry->subsystem);
+				_OutputStream.writeCharacters("\n");
+				_OutputStream.writeCharacters("2CICGRPINFO    cgroup name : ");
+				_OutputStream.writeCharacters(cgEntry->cgroup);
+				_OutputStream.writeCharacters("\n");
+				OMRCgroupMetricElement metricElement = {0};
+				OMRCgroupMetricIteratorState cgroupState = {0};
+				rc = omrsysinfo_cgroup_subsystem_iterator_init(cgEntry->flag, &cgroupState);
+				if (0 == rc) {
+					if (0 != cgroupState.numElements) {
+						/* print the data in columns */
+						while (omrsysinfo_cgroup_subsystem_iterator_hasNext(&cgroupState)) {
```

Comment:
The condition to the `while` loop should be a binary condition, something like `... != 0`

Reply:
How are returns indicating errors handled?  I thought the OMR PR suggested the loop should continue?

Reply:
> The condition to the `while` loop should be a binary condition, something like `... != 0`

Added the binary condition in `while` loop 
`while (0 != omrsysinfo_cgroup_subsystem_iterator_hasNext(&cgroupState))`

> How are returns indicating errors handled? I thought the OMR PR suggested the loop should continue?

Currently we are not printing the metric if there is any error. The iterator continues and prints the available metrics.